### PR TITLE
[MM-56986] imports/import_validators: add valid guest role check

### PR DESCRIPTION
**Pull request by isacikgoz**
#### Summary
Adds a sanity check if the user roles are consistent with the current "Guest" role scheme. The current scheme mandate a user has to be a guest all over the hierarchy.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-56986

#### Release Note

```release-note
NONE
```

### CODE REVIEW

File: server/channels/app/imports/import_validators.go [hunk 1]
```diff
@@ -650,3 +654,37 @@ func isValidEmailBatchingInterval(emailInterval string) bool {
 		emailInterval == model.PreferenceEmailIntervalFifteen ||
 		emailInterval == model.PreferenceEmailIntervalHour
 }
+
+// isValidGuestRoles checks if the user has both guest roles in the same team or channel.
+// at this point we assume that the user has a valid role scheme.
+func isValidGuestRoles(data UserImportData) bool {
+	var isSystemGuest bool
+	if data.Roles != nil {
```
Comment (agnivade) [inline server/channels/app/imports/import_validators.go:662]: This check is already done above.

File: server/channels/app/imports/import_validators.go [hunk 2]
```diff
@@ -650,3 +654,37 @@ func isValidEmailBatchingInterval(emailInterval string) bool {
...
+		isSystemGuest = model.IsInRole(*data.Roles, model.SystemGuestRoleId)
+	}
+
+	var isTeamGuest, isChannelGuest bool
+	if data.Teams != nil {
+		for _, team := range *data.Teams {
+			if team.Roles != nil {
+				isTeamGuest = model.IsInRole(*team.Roles, model.TeamGuestRoleId)
```
Comment (agnivade) [inline server/channels/app/imports/import_validators.go:670]: What if the user is part of two teams, and a guest in one team and not in the other? Then the variable will become true, and then false again in the next iteration. Is that intended? Same for channels.

Comment (isacikgoz) [reply server/channels/app/imports/import_validators.go:670]: From the ticket, what I understood is a guest is always a guest.

> In Mattermost, guests are always guests. That is, if they are a guest in one channel or team, they are a guest in all channels and teams. Unfortunately, we don’t have a single source of truth when it comes to guests. That is…they must be guests at the System Level, guest at the Team Level and guests at the Channel level.

cc @sbishel

Comment (isacikgoz) [reply server/channels/app/imports/import_validators.go:670]: Yes the role for any given entity (system level, team or channel level) has to be guest if a user is a guest. So the fix for the logic can be also made to check if there is inconsistency across teams/channels, will take a look into that. Good catch!

File: server/channels/app/imports/import_validators.go (hunk 1 above)
Comment (isacikgoz) [reply server/channels/app/imports/import_validators.go]: Right, maybe we should avoid that check in the upper level, seems like we are okay with empty roles in the UserImportData afterall.

**Review by agnivade — APPROVED** [review]
Thanks!

**Pull request merged**